### PR TITLE
Feature/rw edit details

### LINKED ISSUE #40: On form submit, update profile information
On the new form view, when the form submit button is pressed:

1. Make api call to back-end to post the new user details (image may need a separate call since it's a different endpoint?) 
2. We should get back the updated details object, and then dispatch that to the store
3. The above will involve making a new action creator, and then adding a new case to the user reducer to handle that action & update the store accordingly with the updated user details

### PR DESCRIPTION
**Pull request by Abdeboskey**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
* Add onSubmit functionality to editProfile.js
* Connected FE to BE
* Create new action creators/reducers to connect state to the store.
* Fixed issue with uploading image

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:

* This has been tested in the browser using the redux devtools, visual confirmation and the React devtools.

# Message for reviewer:

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

make changes to future past

### CODE REVIEW

File: code/web/src/modules/user/EditProfile.js [hunk 1]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
-  //   event.preventDefault()
-    
-  //   this.setState({
-  //     isLoading: true,
-  //   })
+  onSubmit = (event) => {
+    event.preventDefault()
+
+    let newProfileData = this.state.newProfileData
+    newProfileData.id = this.props.user.details.id
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:61]: Smart idea to fold the id property into our newProfileData object

File: code/web/src/modules/user/EditProfile.js [hunk 2]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
+        if (this.props.user.error !== '') {
+          this.props.messageShow(this.props.user.error)
+        } else {
+          this.props.messageShow('Information saved successfully.')
+          //might need something else here
+          //might need to save image path on user
+        }
+      })
+      .catch((error) => {
+        this.props.messageShow('There was some error. Please try again.')
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:85]: Hopefully we can figure out why this error message keeps showing when the info actually gets saved successfully

File: code/web/src/modules/user/EditProfile.js [hunk 3]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
+
+    this.setState({
+      isLoading: true,
+    })
+
+    let data = new FormData()
+    data.append('file', event.target.files[0])
+
+    // Upload image
+    this.props.upload(data)
```
Comment (rwilliams659) [inline code/web/src/modules/user/EditProfile.js:109]: So cool we could re-use this upload function!

File: code/web/src/modules/user/Profile.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ import { level1 } from '../../ui/common/shadows'
 // App Imports
 import userRoutes from '../../setup/routes/user' 
 import { logout } from './api/actions'
-import { APP_URL } from '../../setup/config/env'
+import { APP_URL, APP_URL_API } from '../../setup/config/env'
```
Comment (mlynch5187) [inline code/web/src/modules/user/Profile.js:18]: Really impressed you got this figured out so quickly.

File: code/web/src/modules/user/api/actions.js [hunk 5]
```diff
@@ -66,6 +68,37 @@ export function login(userCredentials, isLoading = true) {
   }
 }
 
+export function updateProfileInfo(newProfileData) {
+  return dispatch => {
+
+    return axios.post(routeApi, mutation({
+      operation: 'userUpdate',
+      variables: newProfileData,
+      fields: ['id', 'name', 'email', 'bio', 'street', 'city', 'state', 'zip', 'image']
```
Comment (rwilliams659) [inline code/web/src/modules/user/api/actions.js:77]: Love how easily this hooked into the BE to update data!

**Review by rwilliams659 — APPROVED** [review]
Great job team, we're chipping away at this.

File: code/web/src/modules/user/EditProfile.js [hunk 6]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
+        this.props.messageShow('There was some error. Please try again.')
+
+        this.setState({
+          isLoading: false,
+        })
+      })
+      .then(() => {
+        window.setTimeout(() => {
+          this.props.messageHide()
+        }, 5000)
```
Comment (Strompy) [inline code/web/src/modules/user/EditProfile.js:94]: I love the timed message pop up!

File: code/web/src/modules/user/EditProfile.js [hunk 7]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
-  // onSubmit = (event) => {
-  //   event.preventDefault()
-    
-  //   this.setState({
-  //     isLoading: true,
-  //   })
+  onSubmit = (event) => {
+    event.preventDefault()
+
+    let newProfileData = this.state.newProfileData
```
Comment (JoshSevy) [inline code/web/src/modules/user/EditProfile.js:60]: I think we will be able to thin this on submit function love that we are reusing most of the projects functionality

File: code/web/src/modules/user/EditProfile.js [hunk 8]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
+      })
+      .catch((error) => {
+        this.props.messageShow('There was some error. Please try again.')
+
+        this.setState({
+          isLoading: false,
+        })
+      })
+      .then(() => {
+        window.setTimeout(() => {
```
Comment (JoshSevy) [inline code/web/src/modules/user/EditProfile.js:92]: I love the use of the setTimeouts

File: code/web/src/modules/user/EditProfile.js [hunk 9]
```diff
@@ -53,84 +54,90 @@ class EditProfile extends Component {
...
+          this.props.messageShow('File uploaded successfully.')
+
+          let image = this.state.image
+          image = `${response.data.file}`
+
+          let newProfileData = this.state.newProfileData
+          newProfileData.image = image
+
+          this.setState({
+            newProfileData
```
Comment (Strompy) [inline code/web/src/modules/user/EditProfile.js:121]: The image upload is so cool! Glad that is all working

**Review by mlynch5187 — APPROVED** [review]
Great work researching this and getting it working in such a short timeframe!

File: code/web/src/modules/user/api/actions.js (hunk 5 above)
Comment (sienna-kopf) [reply code/web/src/modules/user/api/actions.js:77]: This is so cool!! Love seeing how the two intersect!

File: code/web/src/modules/user/EditProfile.js (hunk 3 above)
Comment (Strompy) [inline code/web/src/modules/user/EditProfile.js:109]: I'm curious about industry practices with uploading and storing images (and other files types)

File: code/web/src/modules/user/EditProfile.js [hunk 10]
```diff
@@ -181,7 +189,7 @@ class EditProfile extends Component {
               <Input
                 type="text"
                 fullWidth={true}
```
Comment (JoshSevy) [inline code/web/src/modules/user/EditProfile.js:191]: I think we can add pattern and some regex for form validation

File: code/web/src/modules/user/EditProfile.js (hunk 1 above)
Comment (sienna-kopf) [reply code/web/src/modules/user/EditProfile.js:61]: This all seems really interesting! I hope we have time at some point for a walkthrough.

**Review by Strompy — APPROVED** [review]

**Review by JoshSevy — APPROVED** [review]
Great work team! So happy I got all of you as my team.

**Review by sienna-kopf — APPROVED** [review]
Nice work yall!!! You are crushing this!

**Pull request merged**